File: specs/typespec/admin/marketplace.go
package admin

// Admin marketplace types are defined in vetchium-api-server.typespec/org.
// Admin marketplace handlers import from that package directly using:
//
//	import orgspec "vetchium-api-server.typespec/org"
//
// The types used by admin marketplace routes are:
//   orgspec.AdminCreateCapabilityRequest, orgspec.AdminUpdateCapabilityRequest
//   orgspec.AdminListListingsRequest, orgspec.AdminListListingsResponse
//   orgspec.AdminSuspendListingRequest, orgspec.AdminReinstateListingRequest
//   orgspec.AdminListSubscriptionsRequest, orgspec.AdminListSubscriptionsResponse
//   orgspec.AdminCancelSubscriptionRequest
//   orgspec.MarketplaceListing, orgspec.MarketplaceCapability, orgspec.MarketplaceSubscription

File: specs/typespec/admin/marketplace.ts
// Admin marketplace types are co-located in org/marketplace.ts.
// Org-side approve/reject (OrgApproveListingRequest, OrgRejectListingRequest) are
// intentionally excluded — those are intra-org operations, not admin portal operations.
export type {
	MarketplaceCapability,
	ListCapabilitiesResponse,
	MarketplaceListing,
	AdminCreateCapabilityRequest,
	AdminUpdateCapabilityRequest,
	AdminListListingsRequest,
	AdminListListingsResponse,
	AdminSuspendListingRequest,
	AdminReinstateListingRequest,
	MarketplaceSubscription,
	AdminListSubscriptionsRequest,
	AdminListSubscriptionsResponse,
	AdminCancelSubscriptionRequest,
	CapabilityStatus,
	MarketplaceListingStatus,
	MarketplaceSubscriptionStatus,
} from "../org/marketplace";
export {
	validateAdminCreateCapabilityRequest,
	validateAdminUpdateCapabilityRequest,
	validateAdminListListingsRequest,
	validateAdminSuspendListingRequest,
	validateAdminReinstateListingRequest,
	validateAdminListSubscriptionsRequest,
	validateAdminCancelSubscriptionRequest,
} from "../org/marketplace";
